### ISSUE TITLE
TypeError: Cannot set property 'parentEl' of undefined

### ISSUE THREAD

**Issue by TrejGun** [skratchdot/react-bootstrap-daterangepicker#165]
i have hybrid react (16.2) app
with v3.4.1 it was working good

```jsx
<DateRangePicker
	timePicker
	timePicker24Hour
	showDropdowns
	locale={locale}
	startDate={this.state.startDate}
	endDate={this.state.endDate}
	autoApply
	onEvent={::this.handleEvent}
>
	<InputGroup>
		<Input value={`${start}${rangeSeparator}${end}`} />
		<InputGroup.Button>
			<Button className="default date-range-toggle">
				<Glyphicon glyph="calendar" />
			</Button>
		</InputGroup.Button>
	</InputGroup>
</DateRangePicker>
```

but with 4.0.1 it throws error

```
TypeError: Cannot set property 'parentEl' of undefined
at DateRangePicker (~/node_modules/react-bootstrap-daterangepicker/dist/bundle.js:79:23)
at resolve (~/node_modules/react-dom/cjs/react-dom-server.node.development.js:2102:14)
at ReactDOMServerRenderer.render (~/node_modules/react-dom/cjs/react-dom-server.node.development.js:2260:22)
at ReactDOMServerRenderer.read (~/node_modules/react-dom/cjs/react-dom-server.node.development.js:2234:19)
at renderToString (~/node_modules/react-dom/cjs/react-dom-server.node.development.js:2501:25)
at renderInitialMarkup (~/server/shared/utils/render.jsx:22:37)
at renderAppToString (~/server/shared/utils/render.app.js:47:57)
at renderFE (~/server/shared/routes/fe.js:22:36)
at Layer.handle [as handle_request] (~/node_modules/router/lib/layer.js:93:5)
at trim_prefix (~/node_modules/router/index.js:335:13)
at ~/node_modules/router/index.js:299:7
at Function.process_params (~/node_modules/router/index.js:354:12)
at next (~/node_modules/router/index.js:290:10)
at Layer.handle [as handle_request] (~/node_modules/router/lib/layer.js:89:12)
at trim_prefix (~/node_modules/router/index.js:335:13)
at ~/node_modules/router/index.js:299:7
at Function.process_params (~/node_modules/router/index.js:354:12)
at next (~/node_modules/router/index.js:290:10)
at ~/node_modules/router/index.js:676:15
at next (~/node_modules/router/index.js:275:14)
at Function.handle (~/node_modules/router/index.js:189:3)
at router (~/node_modules/router/index.js:64:12)
```

**Comment by TrejGun** [comment]
i believe the reason is broken export
you should export react component on line 
https://github.com/skratchdot/react-bootstrap-daterangepicker/blob/master/dist/bundle.js#L1939
but instead you export jQuery plugin on line
https://github.com/skratchdot/react-bootstrap-daterangepicker/blob/master/dist/bundle.js#L70

**Comment by TrejGun** [comment]
yep i was able to fix it by adding
```
module.exports = DateRangePicker;
```
as the last statement in file

please fix this

**Comment by TrejGun** [comment]
probably the easies fix id to replace 

```js
export default DateRangePicker;
```

with

```js
module.exports = DateRangePicker;
```

in https://github.com/skratchdot/react-bootstrap-daterangepicker/blob/master/lib/index.js

**Comment by smandegar** [comment]
👍 Can confirm that I am experiencing the same issue

**Comment by skratchdot** [comment]
Thank you all for catching this (and sorry).  I haven't actually used this lib in a while, but i tried setting up some actual unit tests finally, and starting to get code coverage (even though the coverage sucks right now).  I'll look into this tonight, or merge a PR.  I also played with using rollup to create the dist file (which i hadn't used before)

**Comment by skratchdot** [comment]
I also plan on adding some more storybook examples...

**Comment by skratchdot** [comment]
The rollup config is generating: 
```
exports.DateRangePicker = DateRangePicker;
exports.default = DateRangePicker;
```

So you might be able to:
```
const DateRangePicker = require('react-bootstrap-daterangepicker').DateRangePicker;
```

I need to spend a little time looking into this, and updating the README again. Thanks for the report.

**Comment by TrejGun** [comment]
this is not really related to SSR. so nevermind.
i'd like to ask you to not bundle bootstrap-daterangepicker into your package at all
can you please put it to peerDeps?

**Comment by suecom** [comment]
Tried the require, but didn't work-around. Not sure what the equivalent import is? I import like this:

import DateRangePicker from 'react-bootstrap-daterangepicker';

but am getting the same 'TypeError: Cannot set property 'parentEl' of undefined' when extending the component such:

return (
      <DateRangePicker ...>
        
      </DateRangePicker>
    );

Using 3.4.1 in the meantime. Thanks for your work...

**Comment by smandegar** [comment]
Workaround didn't work for me so I've downgraded back to 3.4.1 in the meantime as well

**Comment by skratchdot** [comment]
Hey- thanks for all the feedback in this thread.  I misread the stacktrace `react-dom-server.node` which made me think it was due to server-side rendering.  I'm guessing it's due to the `<InputGroup>` component. @TrejGun - can I see the render method of that component?

To support react 16 correctly, I need to handle arrays and fragments correctly I'm guessing.

When I first wrote this component years ago, I always wrapped the children in a `<div />` like this:
https://github.com/skratchdot/react-bootstrap-daterangepicker/blob/7c67a8d7c14dfc571836c1359b3922456cf6d083/lib/index.js#L111

But at some point, people wanted the ability to not have that wrapper div, so now the code can do:
https://github.com/skratchdot/react-bootstrap-daterangepicker/blob/7c67a8d7c14dfc571836c1359b3922456cf6d083/lib/index.js#L108-L109

I'm guessing mounting does not always work correctly in that case (with custom components).  Perhaps I need to only mount if there is 1 child, and it is a valid html element that is compatible with the `bootstrap-daterangepicker` I wrap...

**Comment by skratchdot** [comment]
@TrejGun - I wish I would've looked into it closer.  You are correct: I am just going to make `bootstrap-daterangepicker` a peer dependency.  That makes a lot more sense.  I hope the other changes I made don't mess anyone up, but I did go ahead and always include a wrapper div (b/c that seems to make more sense instead of injecting into random components. I've added props to allow overriding the container className and style.  If that change causes issues for people, I can revert it.

I hope the peer dependency changes work for everyone!

See the updated instructions: https://github.com/skratchdot/react-bootstrap-daterangepicker#getting-started

The new version is `4.1.0`.  I know that's not really following semver, but it feels weird to have a v4 that never worked.
